braintree.go

```diff
@@ -18,14 +19,24 @@ const (
 	apiVersion4            = 4
 )
 
+const defaultTimeout = time.Second * 60
+
+var defaultClient = &http.Client{Timeout: defaultTimeout}
+
 // New creates a Braintree with API Keys.
 func New(env Environment, merchId, pubKey, privKey string) *Braintree {
-	return &Braintree{credentials: newAPIKey(env, merchId, pubKey, privKey)}
+	return &Braintree{
```

Comment:
I have not written Go seriously in a while so this may be convention, but calling `NewWithHttpClient` with `defaultClient` here would reduce duplication ever so slightly